### PR TITLE
Updating README to refer to ANTLR 5

### PR DESCRIPTION
**Pull request by ftomassetti**
I took a first look at the README, renaming most references to ANTLR 4 into references to ANTLR 5.

I also tried to:
* State that this project is experimental, and that most people should look at ANTLR 4
* Summarizing very shortly why ANTLR 5 came to by and pointing out to antlr5-specs for the details
* Remove some links that were very specific to ANTLR 4
* Add a comment about The Definitive ANTLR 4 Reference, in the context of ANTLR 5
* Add a reference to grammars-v5 (while keeping a reference to grammars-v4)

This is related to #3 

Follow up: Update image to change master into main. This would be easier if we had the sources of the image.

### CODE REVIEW

**Comment by ericvergnaud** [comment]
Thanks! I'm fine with the proposed changes, but @KvanTTT has asked to put on hold any PR until he completes #6 so I'll merge once done.

**Comment by ftomassetti** [comment]
Makes sense, thank you Eric!

**Comment by sschuberth** [comment]
> put on hold any PR until he completes #6 so I'll merge once done.

That PR doesn't touch the `README.md`, though... yet.

**Comment by ericvergnaud** [comment]
Sure but he might need to rebase to an old commit, in which case we'd have to redo all this...
Let's be patient, it looks like we're not very far...

File: README.md
Comment (kaby76) [inline README.md]: The phrase "without actions" has bothered me for a while. I don't think it is correct.

First, there are many grammars in grammars-v4 that contain both "actions" and "semantic predicates". According to the documentation:

* ["Actions are blocks of text written in the target language and enclosed in curly braces. The recognizer triggers them according to their locations within the grammar."](https://github.com/antlr/antlr4/blob/master/doc/actions.md#actions-and-attributes)

* ["Semantic predicates, {...}?, are boolean expressions written in the target language that indicate the validity of continuing the parse along the path "guarded" by the predicate."](https://github.com/antlr/antlr4/blob/master/doc/predicates.md#semantic-predicates)

In either case, we have a slew.
* "actions", using Trash, `//actionBlock[not(following-sibling::QUESTION)]`. 
[actions.txt](https://github.com/antlr/antlr5/files/14054023/actions.txt)

* "semantic predicates", using Trash, `//actionBlock[following-sibling::QUESTION]`. 
[sem-preds.txt](https://github.com/antlr/antlr5/files/14054030/sem-preds.txt)

Secondly, they are lexed identically, and parsed nearly identically.

The phrase should be deleted.

Comment (ftomassetti) [reply README.md]: Thank you, I will remove the sentence

**Comment by KvanTTT** [comment]
@ftomassetti please rebase on `dev` and remove duplicated commit (78a857c4d0b9b7c48713e15499c550368f5ab058).

**Comment by KvanTTT** [comment]
> That PR doesn't touch the README.md, though... yet.

Yes, but I want to keep git history as clear as possible. Maybe it's out of priority for other contributors but it's important for me (I've been working on Kotlin Compiler for almost 3 years and I understand how it could be important). That's why I adhere `Rebase and merge` or `Squash and merge` merging methods that don't produce extra merging commits.

And I didn't want to have extra commits in `dev` before #6 merging (I can't interactively rebase that branch because it is non-standard since it contains merge from unrelated histories). Yes, I could use commits cherry-picking, but it was better to wait until merging.

Now the merge is finished (force-push to dev) and other PRs can be merged using safe GitHub methods.

**Comment by ftomassetti** [comment]
I know I need to re-sign the commits, but I am currently at a computer where I need to reconfigure GPG, so I will do that as soon as I can

**Comment by ftomassetti** [comment]
Ok, it seems now that DCO and tests are all green

**Comment by ericvergnaud** [comment]
I suggest you rebase, I had some changes on my side not yet merged.

**Comment by ftomassetti** [comment]
Sure, I have now rebased

File: README.md
Comment (ericvergnaud) [inline README.md]: maybe point to the antlr web site rather than the GH repo ?

File: README.md
Comment (ericvergnaud) [inline README.md:59]: Good stuff 👍

File: README.md
Comment (ericvergnaud) [inline README.md]: unwanted LF/CR ?

File: README.md
Comment (ericvergnaud) [inline README.md]: how about `grammars verified for ANTLR 4` ?

**Review by ericvergnaud — CHANGES_REQUESTED** [review]

File: README.md
Comment (ftomassetti) [reply README.md]: At this time antlr4.org redirects to antlr.org. In the future antlr.org could mention antlr5, so I thought it could make sense to point specifically to that repository. But it also true that maybe in the future antlr4.org would be a different site that antlr.org, so I will update the link

File: README.md
Comment (ftomassetti) [reply README.md:59]: Thanks!

File: README.md
Comment (ftomassetti) [reply README.md]: Right, removing

File: README.md
Comment (ftomassetti) [reply README.md]: changed

File: README.md
Comment (ftomassetti) [reply README.md]: Marking it as done, please reopen if you want me to point to antlr.org directly

**Review by ericvergnaud — APPROVED** [review]

**Comment by ftomassetti** [comment]
Thank you for approving the PR!
I think I cannot merge this myself, so would you merge it?
I see there is also a check failing (regarding the kotlin runtime but only on macos). However, given I just touched the README, I think we can safely consider it unrelated to the PR

**Pull request merged**
